### PR TITLE
LPS-132752 clay:label tag should allow for literal values

### PR DESCRIPTION
**Pull request by diegonvs**
This PR came originally from https://github.com/liferay-frontend/liferay-portal/pull/1136

There were some weird CI failings and some rebasing were needed and to see the new CI results I created this PR

### CODE REVIEW

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LabelTag.java [hunk 1]
```diff
@@ -56,6 +56,10 @@ public boolean getLarge() {
 		return _large;
 	}
 
+	public boolean getTranslateLabel() {
```
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LabelTag.java:59]: Naming is hard™ .  A standard naming convention for a boolean getter is to have `is` prefix, so this would be `isTranslateLabel`. This sounds awkward, grammatically incorrect, implying that a better name would an _adjective_ rather than an _action_. This convention makes a lot of sense, as an adjective can be true of false. For example, `disabled` is true or false. An attribute with action name implies that its value is implementation of that action, i.e. a function `disable()`.

In short, we should rename this attribute :slightly_smiling_face: . I'd suggest `translated`, with `isTranslated` getter. We can drop `label`, because tag has only one translatable string. Something like `translatedLabel` would maybe make sense in a different tag that has something like `translatedTitle`, and we want to provide API for translation opt-in on either.

On a related note, we can safely rename `getLarge` to `isLarge`.

**Review by markocikos — APPROVED** [review]
@diegonvs  A non-blocking naming nit inline. Other than that, LGTM!

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LabelTag.java (hunk 1 above)
Comment (bryceosterhaus) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LabelTag.java:59]: @markocikos I remember vaguely that we used to not used `is*` prefixes. Do you remember anything like that in the past? I feel like it came from bchan and the java side of formatting, but I don't fully remember. I did search through DXP and we do use it and I am personally not against using it but I remember being steered away from them in the past. 🤷

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:relevant

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LabelTag.java (hunk 1 above)
Comment (diegonvs) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LabelTag.java]: When changing from `getLarge` -> `isLarge` and `getTranslated` -> `isTranslated` it requires a major version bump on `frontend-taglib-clay`. Considering it, does make sense just bump it? From https://github.com/liferay-frontend/liferay-portal/pull/1171#issuecomment-866411456 @nhpatt @bryceosterhaus @markocikos I'm not a pro in DXP semvar but in other packages it looks strong. Should I just place a deprecation notice in `getLarge` and see if semvar cries?

Comment (julien) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LabelTag.java]: @bryceosterhaus for recent tags ([Cards](https://github.com/liferay/liferay-portal/blob/4b9d2320ae42d5c58a56e0bced49f8fe4e7b7bbc/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/BaseCardTag.java#L141-L175), [Management Toolbar](https://github.com/liferay/liferay-portal/blob/4b9d2320ae42d5c58a56e0bced49f8fe4e7b7bbc/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/ManagementToolbarTag.java#L307-L399)), I know that we've used that `is*` (if you look at those methods: for cardsprefix and @markocikos 's explanation makes sense. As Marko said, sometimes it sounds correct (e.g. `isDisabled`) and sometimes it seems *totally* incorrect (e.g. `isShowCreationMenu`). I would think that "native english speakers" could be shocked about this but apparently it's not the case :shrug: 

@diegonvs I'd follow @markocikos 's advice - what I don't really like is marking `getLarge` as deprecated and adding `isLarge` which does exactly the same thing but we've already done that too in various places in order to follow this `is*` convention; I guess it's not that bad.

**Comment by adolfopa** [comment]
@diegonvs You need to run baseline in the module after your latest changes.

**Comment by diegonvs** [comment]
ci:forward

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by julien** [comment]
@diegonvs just so you know, at least one of the errors is the same as for #1179

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LabelTag.java (hunk 1 above)
Comment (markocikos) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LabelTag.java]: > I remember vaguely that we used to not used `is*` prefixes. Do you remember anything like that in the past? I feel like it came from bchan and the java side of formatting, but I don't fully remember. I did search through DXP and we do use it and I am personally not against using it but I remember being steered away from them in the past.

@bryceosterhaus If I recall correctly, there is a bug in Soy related to getter prefix, when you wire Java to Soy. If you pass in a custom POJO as a param to Soy template, you can access it's boolean property only if it has a `get` prefix. I might be misremembering, I ran into this several years ago, in my LCS days, on DXP 7.0. This issue can be safely ignored for this PR. It is only applicable if you are creating a model POJO used by a Soy template.

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
> @diegonvs just so you know, at least one of the errors is the same as for #1179

Thanks for the info Julien <3

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
Screen Shot 2021-07-06 at 13 05 39



> LocalFile.ClaySelect#SmokeRegularSelect 
It's unrelated. Re-running again.

**Comment by diegonvs** [comment]
ci:forward

**Comment by diegonvs** [comment]
`org.openqa.selenium.JavascriptException: javascript error: Failed to execute 'elementsFromPoint' on 'Document': The provided double value is non-finite.` as https://github.com/liferay-frontend/liferay-portal/pull/1179 @zhangyan0701 @nkkwon

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
Manually sent in https://github.com/brianchandotcom/liferay-portal/pull/104247
